lsp-methods.el

```diff
@@ -879,7 +879,19 @@ to a text document."
       :annotation-function #'lsp--annotate
       :display-sort-function #'lsp--sort-completions)))
 
-;;; TODO: implement completionItem/resolve
+(defun lsp--resolve-completion (comp)
+  (lsp--cur-workspace-check)
+  (lsp--send-changes lsp--cur-workspace)
+  (unless (gethash "resolveProvider" (lsp--capability "completionProvider"))
+    (user-error "This language server doesn't support resolving completion items"))
```

Comment:
Can you simply return the completion item? Callers shouldn't worry about it.

Reply:
Like the current one? Yes, that is indeed one possibility.

Reply:
Consider writing a function `lsp--get-completion-doc`. One may write:

```elisp
(defun lsp--get-completion-doc (item)
  (or (gethash "documentation" item)
      (gethash "documentation" (lsp--resolve-completion item)))
```

This will throw an error if the server doesn't support resolving completion items. To write an error-free implementation, it has to either catch the error, or check `resolveProvider` before calling.

Simply return `item` when `resolveProvider` is nil can avoid that.

Reply:
Ah yes, I did do that on that new PR of mine (because this one was being quite silly indeed and wouldn't update after this particular change).

Do you recon that we should also return a nil completion item if someone passes us nil?

Reply:
Will comment on the new PR.